src/main/java/core/basesyntax/dao/jdbc/DaoJdbcManufacturerImpl.java

```diff
@@ -0,0 +1,126 @@
+                item.setId(resultSet.getObject(1, Long.class));
+            }
+            statement.close();
+        } catch (SQLException throwables) {
+            throw new DataProcessingException("ERROR: create() method has failed for "
+                    + item, throwables);
+        }
+        return item;
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        String query = "SELECT * FROM manufacturers WHERE manufacturer_id = ? AND "
+                + "manufacturer_deleted = false";
+        Manufacturer manufacturer = null;
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            statement.close();
+            if (resultSet.next()) {
+                manufacturer = parseResultSet(resultSet);
+            }
+        } catch (SQLException throwables) {
+            throw new DataProcessingException("ERROR: get() method has failed for id="
+                    + id, throwables);
+        }
+        return Optional.ofNullable(manufacturer);
+    }
+
+    @Override
+    public List<Manufacturer> getAll() {
+        String query = "SELECT * FROM manufacturers WHERE manufacturers_deleted = false";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            ResultSet resultSet = statement.executeQuery();
+            statement.close();
+            List<Manufacturer> list = new ArrayList<>();
+            while (resultSet.next()) {
+                list.add(parseResultSet(resultSet));
+            }
+            return list;
+        } catch (SQLException throwables) {
+            throw new DataProcessingException("ERROR: getAll() method has failed", throwables);
+        }
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer item) {
+        if (get(item.getId()).isPresent()) {
+            String query = "UPDATE manufacturers SET manufacturer_name = ?, "
+                    + "manufacturer_country = ? WHERE manufacturer_id = ?";
+            try (Connection connection = ConnectionUtil.getConnection()) {
+                PreparedStatement statement = connection.prepareStatement(query);
+                statement.setString(1, item.getName());
+                statement.setString(2, item.getCountry());
+                statement.setLong(3, item.getId());
+                statement.executeUpdate();
+                statement.close();
+                return get(item.getId()).get();
+            } catch (SQLException throwables) {
+                throw new DataProcessingException("ERROR: update() method has failed for "
+                        + item, throwables);
+            }
+        }
+        throw new RuntimeException("ERROR: Database contains no " + item + " object to update");
```

Comment:
we can remove this exception

Reply:
Can I catch this and just do sout? Because absence of object crashing my program

Reply:
no, we shouldn't use sout. RuntimeException is ok